lisa/analysis/load_tracking.py

```diff
@@ -230,7 +228,9 @@ def df_top_big_tasks(self, util_threshold, min_samples=100):
         samples = samples.sort_values(ascending=False)
 
         top_df = pd.DataFrame(samples).rename(columns={"util" : "samples"})
-        top_df["comm"] = top_df.index.map(self.trace.get_task_by_pid)
+        def get_name(pid):
+            return self.trace.get_task_pid_names(pid)[-1]
```

Comment:
`get_task_pid()` (which should be used by most analysis functions ATM) returns `pid_list[0]` AKA the first PID.
Depending on what you're looking at, either makes sense, but it might better to keep that change for a separate PR.

Reply:
Looks like you confused `get_task_by_pid()` with`get_task_pid()` :)

Regarding `get_task_pid()`, it actually returns the last PID of the first task that ran with the given name,  because of `self._tasks_by_pid = (df.drop_duplicates(subset=pid_key, keep='last')`
So if you had:
```
PID NAME
1 sh
2 sh
2 mycommand
2 mycommand_renamed_itself
3 sh
3 mycommand2
3 mycommand_renamed_itself
```
You would get:
```
# API deprecated by this PR
get_task_pid("mycommand_renamed_itself") == 2 # interesting but irrelevant in our case
get_task_by_pid(2) == "mycommand_renamed_itself" # this one is relevant for this comment

# New API
get_task_pid_names(2)[-1] == "mycommand_renamed_itself"
``

So here we can see that `get_task_by_pid(pid)` can be correctly replaced by `get_task_pid_names(pid)[-1]`.

Reply:
> Looks like you confused `get_task_by_pid()` with`get_task_pid()` :)
> 

No, but I did forget about the `drop_duplicates(..., keep='last)`. It's been there for some time, but I swear I've seen the warning fire at least once in `get_task_pid()`. I don't really know how to make sense of that with the `drop_duplicates()`.

Reply:
Ah ok, I was under that impression since the hunk you commented on does not contain any reference to `get_task_pid()`.

In the example I provided, that should give the warning you talk about:
```
get_task_pid("mycommand_renamed_itself")
```
The `drop_duplicates(pid, keep='last')` will turn the name table into:
```
1 sh
2 mycommand_renamed_itself
3 mycommand_renamed_itself
```
So the `pid_list` in `get_task_pid("mycommand_renamed_itself")` will be set to `[2, 3]`, which will give the warning and return `[2,3][0] == 2`. This choice of returning `2` seems completely arbitrary, as there is nothing more interesting in PID 2 than in 3 (they are different tasks, probably sharing a common goal and even code, but instantiated at two unrelated point in time). That really should raise an exception, which it will do after this pull request (`get_task_pid()` has been re-implemented in terms of `get_task_id()`, which is strict by default).